File: src/array/python3/26_RemoveDuplicatesFromSortedArray.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        length=len(nums)
        max=nums[length-1]
        for i in range(0,length,1):
            if nums[i] != max:
                for j in range(i+1,length,1):
                    if nums[i] < nums [j]:
                        nums[i+1],nums[j]=nums[j],nums[i+1]
                        break
            else:
                break
        return i+1
            
        
# @lc code=end

File: work/typescript/array/26_RemoveDuplicatesFromSortedArray.ts
/*
 * @lc app=leetcode id=26 lang=typescript
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
function removeDuplicates(nums: number[]): number {
    let dump:number=0;
    let i:number=0;
    while(true){
        let j:number=i+1+dump;
        while(j<nums.length){
            if(nums[i]==nums[j]){
                j++;
            }
            else{
                [nums[i+1],nums[j]]=[nums[j],nums[i+1]];
                dump=j-i-1;
                break;
            };
        }
        if(j>=nums.length){
            return i+1;
        };
        i++;
    }
};
// @lc code=end

